Spades fails to compile from source on an aws r7i.2xlarge instance

### Description of bug

We migrated our Jenkins server to an r7i.2xlarge instance (8 vCPU, 64 GiB, DDR5 memory). One of our Jenkins jobs includes building spades from source. Here is an excerpts of the docker file:
```
FROM library/ubuntu:22.04

RUN apt-get update && \
    apt-get upgrade -y && \
    apt-get install -y libbz2-dev liblz1 g++ cmake python-is-python3

RUN git clone --quiet https://github.com/ablab/spades.git -b main &&   \
    cd spades && \
    PREFIX=/usr/local ./spades_compile.sh && \
    cd .. &&   \
    rm -fr spades && \
    spades.py --test
```


### spades.log

[spades.log](https://github.com/user-attachments/files/16764826/spades.log)


### params.txt

no log available. Failure occurs during compilation.

### SPAdes version

SPades version 4.0.0

### Operating System

Ubuntu 22.04

### Python Version

3.10.12

### Method of SPAdes installation

compile from source

### No errors reported in spades.log

- [X] Yes

FWIW this error from the attached log seemed to be the most relevant:

> /spades/ext/include/blaze/blaze/system/CacheSize.h:77:1: error: static assertion failed: Compile time condition violated
>    77 | BLAZE_STATIC_ASSERT( blaze::cacheSize > 100000UL && blaze::cacheSize < 100000000UL );
>       | ^~~~~~~~~~~~~~~~~~~

--- a/ext/include/blaze/blaze/system/CacheSize.h
+++ b/ext/include/blaze/blaze/system/CacheSize.h
@@ -74,7 +74,7 @@ constexpr size_t cacheSize = BLAZE_CACHE_SIZE;
 /*! \cond BLAZE_INTERNAL */
 namespace {
 
-BLAZE_STATIC_ASSERT( blaze::cacheSize > 100000UL && blaze::cacheSize < 100000000UL );
+BLAZE_STATIC_ASSERT( blaze::cacheSize > 100000UL && blaze::cacheSize < UINT64_C(10000000000) );
 
 }
 /*! \endcond */

